setup.md

```diff
@@ -269,8 +269,124 @@ TODO: Explain one of the following:
 * What significance does this have to the user?
 * What's in it for the user? WIIFM (What's in it for me? (for the user))
 
-### To add readline support to Fennel
+The command-line REPL that comes with the `fennel` works out of the box, but
+the built-in line-reader is very limited in user experience. Adding
+[GNU Readline](https://tiswww.case.edu/php/chet/readline/rltop.html) support
+enables such user-friendly features as
+- Tab completion on the REPL that can complete on all locals, macros, and special forms
+- A rolling history buffer, which can be navigated, searched (`ctrl+r`), and optionally
+persisted to disk so you can search input from previous REPL sessions.
+- emacs (default) or vi emulation via readline's custom support for better line
+navigation
+- Optional use of additional readline features in `~/.inputrc`, such as blinking
```

Comment:
Let's link to a reference for `.inputrc` here; when I read this I thought to myself "I wonder what the options for this are; maybe I'm missing out". Edit: I see now it's covered below; maybe mention "see below" here.